### ISSUE TITLE
Add courses to Profile during profile creation

### ISSUE THREAD

**Issue by stevelyall** [binder-app/android#22]
FirstLaunchActivity takes in a class list as a string, and has code to create a Profile object which is sent to the API. Right now courses are not included in profiles.

Profile has an addCourse method and maintains a List of Course objects which is used to convert it to JSON, but we should determine what rules we want to follow parsing course names.

I know we discussed this a lot earlier, should we just take the list of courses as a string from the user? If so, what format are we agreeing to use?

COMP 1130
COMP1130
comp1130
comp 1130
comp 113
or any other misspelling, for that matter?

**Comment by mitchhentges** [comment]
Well, we ideally should have a set list of courses, right? Didn't we decide
that?
That way, a user has to choose from a specified list?

On Sat, Nov 14, 2015 at 10:00 PM, Steve Lyall notifications@github.com
wrote:

> FirstLaunchActivity takes in a class list as a string, and has code to
> create a Profile object which is sent to the API. Right now courses are not
> included in profiles.
> 
> Profile has an addCourse method and maintains a List of Course objects
> which is used to convert it to JSON, but we should determine what rules we
> want to follow parsing course names.
> 
> I know we discussed this a lot earlier, should we just take the list of
> courses as a string from the user? If so, what format are we agreeing to
> use?
> 
> COMP 1130
> COMP1130
> comp1130
> comp 1130
> comp 113
> or any other misspelling, for that matter?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/binder-app/android/issues/22.

## 

[image: Description: transportationGoldBlack.gif]

_Mitchell Hentges_
QA Assistant
Streamline Technologies

direct: 250-374-3831
www.stti.ca

```
        *At Streamline, Safety Is Everyone's Responsibility*
```

---

The information in this email is confidential and may be legally
privileged. It is intended solely for the addressee. Access to this email
by anyone else is unauthorized. If you are not the intended recipient, any
disclosure, copying, distribution or any action taken or omitted to be
taken in reliance on it, is prohibited and may be unlawful. If you received
this in error, please contact the sender.  Streamline Transportation
Technologies Inc.

**Comment by stevelyall** [comment]
Yeah Ideally that would be good, since there are a lot of ways one could mess up entering courses

**Comment by roddicks** [comment]
I like the first format. How are we going to get the course list though? Will TRU give us access to that part of the database?

**Comment by stevelyall** [comment]
We need to create either a static hard-coded list on the client side the user can choose from, or simply make courses a text field. I think @chrisfosterelli  and @mitchhentges are leaning toward putting in a hard coded list.

Here's an idea, since our first deliverable is a prototype, why don't we just make the list of courses a String and let people enter whatever? We don't match on them anyway and this would do fine for now- we can add our list that users can select from for the final mvp next week, if at all. What do you guys think?

**Comment by chrisfosterelli** [comment]
I think that since we are being graded on user experience, not necessarily
application functionality alone, that would be a situation were we may
loose marks if we skimp on it.

**Comment by stevelyall** [comment]
I agree, but the purpose of the current phase is a prototype, at this stage I doubt Park would even bat an eye at us using it as a text field, even though I know we're all not as fussed about it. I think we stand to gain more in his opinion by improving the overall look and feel of the prototype

**Comment by chrisfosterelli** [comment]
In that case, since I haven't seen the state of the prototype app yet I'll
leave that decision up to you guys. But I'll still argue it's really
terrible UX and as a user I would not understand that I was supposed to
enter a list when only provided a standard textbox.

**Comment by chrisfosterelli** [comment]
Out of curiosity though, is there a reason why its difficult to do this?

It looks like it is just a simple ListView with an adapter and the list
coded into the strings file. Could make it even easier by doing it as a sub
activity.

http://theopentutorials.com/tutorials/android/listview/android-multiple-selection-listview/#ListViewMultipleSelectionActivityjava

I may be misunderstanding something though :)

**Comment by roddicks** [comment]
If we're looking at doing the ListView, any selection interface we use is going to need a search implemented because just having a scrolling list will be incredibly hard to use (hundreds of courses to scroll through). 
I feel like unless we implement the search, we might actually lose more marks than any we would lose doing the text box (which I don't think will lose us marks, btw).

**Comment by chrisfosterelli** [comment]
Fair point, I was assuming we were only preloading like 10-20 classes.

**Comment by mitchhentges** [comment]
Why not a listview that filters as you type, only showing courses that contain the text in your search?

**Comment by mitchhentges** [comment]
So, this discussion is mostly at #33.
This issue is done, fixed by #34
